### ISSUE TITLE
Error: The "emitter" argument must be an instance of EventEmitter

### ISSUE THREAD

**Issue by Patrik-Stas** [openwallet-foundation/credo-ts#1751]
Hi folks, I am using `@credo-ts`  modules at version `0.5.0-alpha.101`  + `@hyperledger/*` at version `0.2.0-dev.6`, specifically:
```
    "@credo-ts/anoncreds": "0.5.0-alpha.101",
    "@credo-ts/askar": "0.5.0-alpha.101",
    "@credo-ts/core": "0.5.0-alpha.101",
    "@credo-ts/indy-vdr": "0.5.0-alpha.101",
    "@credo-ts/node": "0.5.0-alpha.101",
    "@hyperledger/anoncreds-nodejs": "^0.2.0-dev.6",
    "@hyperledger/anoncreds-shared": "^0.2.0-dev.6",
    "@hyperledger/aries-askar-nodejs": "^0.2.0-dev.6",
    "@hyperledger/aries-askar-shared": "^0.2.0-dev.6",
    "@hyperledger/indy-vdr-nodejs": "^0.2.0-dev.6",
    "@hyperledger/indy-vdr-shared": "^0.2.0-dev.6",
```

I am trying to receive oob invitation, but when cred attemps to send response, it fails with the error below. Does this rings bells for anyone? Let me know if I can provide some other relevant troubleshooting information.

``` 
ERROR: Error sending message to http://localhost:8081/didcomm/alpha: The "emitter" argument must be an instance of EventEmitter or EventTarget. Received an instance of AbortSignal {
  "error": {
    "name": "TypeError",
    "stack": "TypeError [ERR_INVALID_ARG_TYPE]: The \"emitter\" argument must be an instance of EventEmitter or EventTarget. Received an instance of AbortSignal\n    at Object.fetch (node:internal/deps/undici/undici:11413:11)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async HttpOutboundTransport.sendMessage (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/transport/HttpOutboundTransport.ts:51:20)\n    at async MessageSender.sendToService (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/agent/MessageSender.ts:439:9)\n    at async MessageSender.sendMessage (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/agent/MessageSender.ts:293:9)\n    at async ConnectionsApi.acceptOutOfBandInvitation (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/modules/connections/ConnectionsApi.ts:147:5)\n    at async OutOfBandApi.acceptInvitation (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/modules/oob/OutOfBandApi.ts:584:28)\n    at async OutOfBandApi._receiveInvitation (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/modules/oob/OutOfBandApi.ts:475:14)\n    at async OobController.receiveInvitation (/Users/username/dev/work/projectname/afjproject/src/api/OobController.ts:83:53)",
    "message": "The \"emitter\" argument must be an instance of EventEmitter or EventTarget. Received an instance of AbortSignal",
    "code": "ERR_INVALID_ARG_TYPE"
  },
  "message": "The \"emitter\" argument must be an instance of EventEmitter or EventTarget. Received an instance of AbortSignal",
  "body": {
    "protected": "eyJlbmMiOiJ4Y2hhY2hhMjBwb2x5MTMwNV9pZXRmIiwidHlwIjoiSldNLzEuMCIsImFsZyI6IkF1dGhjcnlwdCIsInJlY2lwaWVudHMiOlt7ImVuY3J5cHRlZF9rZXkiOiJXd3FQZ1pMMnRMNW03dHNidkhvbERCeGtkOWdWbGUzZEVlXzROc1NHR0FqelU0cm1MbDB5Sy1DZTB5VVZBQkFwIiwiaGVhZGVyIjp7ImtpZCI6IjZZNFEyNVFabWY3UUZoc1l6UmFjRm5URmdRRXNEcFRtUUFFd2FTOFR0NkpNIiwic2VuZGVyIjoiOXR0enNzUHlOc2dtSTVneDBDeEQ2X09ZVW9NTHZZMWxSdHBnWVdwMm9SeWNqTmYtUERkVV9WQ3dqenUyeXdwcGpQb1hQVDBVY1dsQll3eHNiN2R4MjZkQi16c0hWeDhNckVHSUtpemZGT2h5bHZSdjFtNUROTVpWYlVvIiwiaXYiOiJlVHdWSDJVREtYbkxncmF1TFNEdWw3ckVKdlh3eTRUcyJ9fV19",
    "ciphertext": "KqU6alzkMemsIERc01wG4sNtQ7Nd3Vw8vKB9DeI4Ic81mpybjO7v2MGAj_QI1hJ12m2Kc8lAyN5fsMi1Fx9JSsmCYH09VTB9iqaHu8ojyp0nraQcWklrKQDNaJKA_4gJf6KaY1LWfuRs_hRz394P-8OlqNWHOSsiFZE18NtjKV-_WLOSynIGePZkQjJP5zGZV0hJk87PpZkcuHBWQyucMsEnJkLorJzKFyrG3bV_gwvlvMcT4lJjO1cqzVubehDSwU0zIfpTWgcoNmzHZIYUY3tyePZf3pK7_W-ssurAcvXFry3yqPC4us2nMew1kku0YQhlzvTK-GwTMjULpghw6bhhS9BSOJRznZOkkIML_UvX7bEpzEqjxggAmJ9S7MDlaq_zUbqLoptS35Q2Z6IEfye3KkfTfVu9iabcycC0ThDhyjo6rYshVX0-py6L5UnSfpqTuQAPbnTIen9OEW42v8v6gKXpZb6UwZneUtBg8zHe--FRB-Bli2E2mOmljifBSbRQCg_LP4u-LzVlLS2cimAdKoscwX1CfYSHACEi0fbaCf7e7Y87HfofzlSpFhPc0TPNftywVyIMRKzcz4garcOMGVUy5uW-82IKHuABOiAnn2qesc_cF2mirT9Awfj4h5SyJlUT83EsXLCEmy3G58gxa9nz4X_ceHhH3FEONNzRHcn92779o35CKa2kHRELMHcMsZ-fEHogagJZXA9spKp5O8J_vphtQcIB4wRfTY_tBLBoLnZ7ppWg3zUu2b0MJnrvkFQIfA88aL8HDeranbunqTB0rAibPaDuaeC7bt6RmNJ1b02xfeS-_yBPROC6PRbk0ktPzkLn9ZQhCr9lhf5Mz5BunGppVy-zWQItyjKwa-EezCWm_d63dW-36hl-jxZDvLpJjHoZKZDtwBrzdthlDnPUlNU-TeddOXUXGXfFaEqiBEpX8GyvYCi_xgNmHXCb6xjv2A4g8VXBEWM1EnAs6c1NQWApU5ZkD2t7SHAa5WKG5HkBzH5uYrIGDq4FfNA31wCEEUXFUqiqXHR-GtNKSeUDvNkGzQpB3FVdUcyqngFMqM0hfC3MwMDdtP2n8FoBjl42EMqgkwYH2kIMp8XdrUhRXvXMEvkET0MD4OyvJvJvYD10eRMwE-fBCtyTakdeaCEX9AdRdqC8hLI_Q_E6T6WX_Kkmte23kCDjqhPJ2hIoxElBlAbfpaynpw",
    "iv": "qNqiCzoMtDcmseWr",
    "tag": "XRwXBaTZHi-aPoXEgLfD7A"
  },
  "didCommMimeType": "application/didcomm-envelope-enc"
}
ERROR: Message is undeliverable to connection aefd916b-aed1-4ad1-82f8-090ddae91ea5 (customlabel) {
  "message": {
    "@type": "https://didcomm.org/connections/1.0/request",
    "@id": "de77a309-b039-436c-ba01-62820c7b5224",
    "label": "foobar",
    "connection": {
      "DID": "5BYgAD1rQepeuhnKUZJ9Tv",
      "DIDDoc": {
        "@context": "https://w3id.org/did/v1",
        "publicKey": [
          {
            "id": "5BYgAD1rQepeuhnKUZJ9Tv#1",
            "controller": "5BYgAD1rQepeuhnKUZJ9Tv",
            "type": "Ed25519VerificationKey2018",
            "publicKeyBase58": "3HCMwmmN1dP2cXWwqyKjpN19BXzizNF4dcVQ4GHVGLwF"
          }
        ],
        "service": [
          {
            "id": "5BYgAD1rQepeuhnKUZJ9Tv#IndyAgentService-1",
            "serviceEndpoint": "didcomm:transport/queue",
            "type": "IndyAgent",
            "priority": 0,
            "recipientKeys": [
              "3HCMwmmN1dP2cXWwqyKjpN19BXzizNF4dcVQ4GHVGLwF"
            ],
            "routingKeys": []
          }
        ],
        "authentication": [
          {
            "publicKey": "5BYgAD1rQepeuhnKUZJ9Tv#1",
            "type": "Ed25519SignatureAuthentication2018"
          }
        ],
        "id": "5BYgAD1rQepeuhnKUZJ9Tv"
      }
    },
    "~thread": {
      "thid": "de77a309-b039-436c-ba01-62820c7b5224",
      "pthid": "42435e92-2b63-4418-94f0-8a3e147993ec"
    },
    "~transport": {
      "return_route": "all"
    }
  },
  "errors": [
    {
      "name": "AriesFrameworkError",
      "message": "Error sending message to http://localhost:8081/didcomm/alpha: The \"emitter\" argument must be an instance of EventEmitter or EventTarget. Received an instance of AbortSignal",
      "stack": "AriesFrameworkError: Error sending message to http://localhost:8081/didcomm/alpha: The \"emitter\" argument must be an instance of EventEmitter or EventTarget. Received an instance of AbortSignal\n    at HttpOutboundTransport.sendMessage (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/transport/HttpOutboundTransport.ts:105:13)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async MessageSender.sendToService (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/agent/MessageSender.ts:439:9)\n    at async MessageSender.sendMessage (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/agent/MessageSender.ts:293:9)\n    at async ConnectionsApi.acceptOutOfBandInvitation (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/modules/connections/ConnectionsApi.ts:147:5)\n    at async OutOfBandApi.acceptInvitation (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/modules/oob/OutOfBandApi.ts:584:28)\n    at async OutOfBandApi._receiveInvitation (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/modules/oob/OutOfBandApi.ts:475:14)\n    at async OobController.receiveInvitation (/Users/username/dev/work/projectname/afjproject/src/api/OobController.ts:83:53)",
      "cause": {
        "name": "TypeError",
        "stack": "TypeError [ERR_INVALID_ARG_TYPE]: The \"emitter\" argument must be an instance of EventEmitter or EventTarget. Received an instance of AbortSignal\n    at Object.fetch (node:internal/deps/undici/undici:11413:11)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async HttpOutboundTransport.sendMessage (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/transport/HttpOutboundTransport.ts:51:20)\n    at async MessageSender.sendToService (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/agent/MessageSender.ts:439:9)\n    at async MessageSender.sendMessage (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/agent/MessageSender.ts:293:9)\n    at async ConnectionsApi.acceptOutOfBandInvitation (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/modules/connections/ConnectionsApi.ts:147:5)\n    at async OutOfBandApi.acceptInvitation (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/modules/oob/OutOfBandApi.ts:584:28)\n    at async OutOfBandApi._receiveInvitation (/Users/username/dev/work/projectname/afjproject/node_modules/@credo-ts/core/src/modules/oob/OutOfBandApi.ts:475:14)\n    at async OobController.receiveInvitation (/Users/username/dev/work/projectname/afjproject/src/api/OobController.ts:83:53)",
        "message": "The \"emitter\" argument must be an instance of EventEmitter or EventTarget. Received an instance of AbortSignal",
        "code": "ERR_INVALID_ARG_TYPE"
      }
    }
  ],
  "connection": {
    "_tags": {},
    "metadata": {},
    "connectionTypes": [],
    "previousDids": [],
    "previousTheirDids": [],
    "id": "aefd916b-aed1-4ad1-82f8-090ddae91ea5",
    "createdAt": "2024-02-08T11:30:50.858Z",
    "did": "did:peer:1zQmW5EeWJwFNoj7gLD8s9uLr13WBC4vF4rPrUMQ4GX4BbDd",
    "invitationDid": "did:sov:BA4a4CCvMuChjWDauUFvpF",
    "theirLabel": "customlabel",
    "state": "request-sent",
    "role": "requester",
    "autoAcceptConnection": true,
    "threadId": "de77a309-b039-436c-ba01-62820c7b5224",
    "protocol": "https://didcomm.org/connections/1.0",
    "outOfBandId": "ae502c53-e4fc-4093-8a80-67d062bcd0af",
    "updatedAt": "2024-02-08T11:30:50.859Z"
  }
}
```

**Comment by genaris** [comment]
I remember a similar problem with an older version of NodeJS 18 and I think it was fixed some time ago (from 18.16 I think, look at https://github.com/nodejs/undici/pull/1910#issuecomment-1500332168). Can you share the exact version you are using and the dependencies your project has?

**Comment by Patrik-Stas** [comment]
@genaris that was it! Thanks - I was using `18.15.0`, the issue is gone with `18.19.0`
